pkg/kubectl/apply/strategy/replace_visitor.go

```diff
@@ -53,7 +55,14 @@ func (v replaceStrategy) MergeType(e apply.TypeElement) (apply.Result, error) {
 
 // MergePrimitive returns a result by merging the recorded, local and remote values
 // - replacing the remote value with the local value
+// If FailOnConflict option is enabled, throw error when conflicts occure in both recorded
+// and remote values.
 func (v replaceStrategy) MergePrimitive(e apply.PrimitiveElement) (apply.Result, error) {
+	if v.options.FailOnConflict && e.HasRecorded() && e.HasRemote() &&
+		fmt.Sprintf("%v", e.GetRecorded()) != fmt.Sprintf("%v", e.GetRemote()) {
+		return apply.Result{Operation: apply.CONFLICT}, fmt.Errorf("Conflict detected in recorded value (%+v) and remote value (%+v)",
```

Comment:
Does this contain the full path to the conflicting value?  If not it will be hard for users to debug.  Don't fix now if it is going to add a bunch of code, but we should at least add a TODO for how to fix this.

Reply:
Good point. I have taken it into consideration. 
primitiveElement is the 'leaf' node for comparing, if conflicts occured, the error message will be thrown out stack. Error message contains the conflict values. In call stack, I want to append the current node with the cached error message recursively, so that the path will be presented.  BTW, the `CONFLICT` is used for identification, without particular error class for conflict by far.